config/resolver.go

```diff
@@ -191,12 +198,30 @@ func (c *configResolver[T]) Resolve(config *T) Validator {
 	}
 }
 
+func (c *configResolver[T]) read(config *T) error {
+	var data []byte
+	var err error
+	if len(c.configFile) > 0 {
+		data, err = c.readFromFile()
+	} else if len(c.data) > 0 {
+		data = c.data
+	}
+	if err != nil {
```

Comment:
We could move it in the if, no? as only readFromFile return err

Reply:
yeah we could do that, but it doesn't change that much to move this check in a deeper condition. Besides since it's a global variable, I think it's better to check it at the root of the function.